ratpack-manual/src/content/chapters/37-pac4j.md

```diff
@@ -0,0 +1,94 @@
+# Security using pac4j
+
+The `ratpack-pac4j` extension provides authentication support based on [pac4j](https://github.com/pac4j/pac4j).
+
+`pac4j` is a generic authentication engine which is implemented by many frameworks and supports multiple protocols: OAuth, CAS, OpenID (Connect), SAML, Google App Engine and HTTP (form and basic auth).
+
+The `RatpackPac4j` class is the main component to use to manage the security of your web application.
+
+
+## Prerequisite
+
+All authentication mechanisms work by redirecting the user to an identity provider and finishing the login process when he is redirected back to the application with some dedicated credentials.
+To make these authentication processes work and to save the originally requested url, the session is required. That's why the `SessionModule` is necessary:
+
+```language-java
+.registry(Guice.registry(b -> b
+                  .module(SessionModule.class)
+```
+
+
+## Authentication mechanisms definition
```

Comment:
I hoped it was clear enoguh: can you elaborate a little more about what's unclear? I don't want to enter into too many details, because for example, for the `FormClient` and the `BasicAuthClient`, I may have added that you need the additional `Authenticator` and `ProfileCreator` objects.